state/state.go

```diff
@@ -101,7 +126,21 @@ func (s *State) Bytes() []byte {
 
 // Mutate state variables to match block and validators
 // after running EndBlock
-func (s *State) SetBlockAndValidators(header *types.Header, blockPartsHeader types.PartSetHeader, prevValSet, nextValSet *types.ValidatorSet) {
+func (s *State) SetBlockAndValidators(header *types.Header, blockPartsHeader types.PartSetHeader, abciResponses *ABCIResponses) {
+
+	// copy the valset
```

Comment:
Fair point. Actually wondering why we need to copy the prevValSet. We need one copy so we can make the updates and then have both the prev and current sets around, but I can't see right away why we need two. @jaekwon